<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from JoeMegaTron.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="JoeMegaTron" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="Joe_body">
    <inertial>
      <mass value="30.0"/>
      <inertia ixx="30.0" ixy="0.0" ixz="0.0" iyy="30.0" iyz="0.0" izz="30.0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.28 0.4 0.24"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.28 0.4 0.24"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="Joe_body">
    <material>Gazebo/Green</material>
  </gazebo>
  <link name="left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.08"/>
      </geometry>
      <origin rpy="0 1.57 0" xyz="0 0 0"/>
    </visual>
    <inertial>
      <mass value="3.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <collision>
      <geometry>
        <cylinder length="0.05" radius="0.08"/>
      </geometry>
      <origin rpy="0 1.57 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="left_wheel">
    <material>Gazebo/Black</material>
    <mu1 value="100.0"/>
    <mu2 value="100.0"/>
    <kp value="1000000.0"/>
    <kd value="1.0"/>
  </gazebo>
  <joint name="left_wheel_axis" type="continuous">
    <limits effort="1000.0" velocity="50.0"/>
    <axis xyz="1 0 0"/>
    <child link="left_wheel"/>
    <parent link="Joe_body"/>
    <origin rpy="0 0 0" xyz="0.105 0.097 -0.14"/>
  </joint>
  <link name="right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.08"/>
      </geometry>
      <origin rpy="0 1.57 0" xyz="0 0 0"/>
    </visual>
    <inertial>
      <mass value="3.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <collision>
      <geometry>
        <cylinder length="0.05" radius="0.08"/>
      </geometry>
      <origin rpy="0 1.57 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="right_wheel">
    <material>Gazebo/Black</material>
    <mu1 value="100.0"/>
    <mu2 value="100.0"/>
    <kp value="1000000.0"/>
    <kd value="1.0"/>
  </gazebo>
  <joint name="right_wheel_axis" type="continuous">
    <limits effort="1000.0" velocity="50.0"/>
    <axis xyz="1 0 0"/>
    <child link="right_wheel"/>
    <parent link="Joe_body"/>
    <origin rpy="0 0 0" xyz="-0.105 0.097 -0.14"/>
  </joint>
  <link name="pivot_box">
    <inertial>
      <mass value="1.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
      <origin rpy="0 0 0" xyz="0 -0.015 -0.025"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.05 0.03 0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 -0.015 -0.025"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.05 0.03 0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 -0.015 -0.025"/>
    </collision>
  </link>
  <gazebo reference="pivot_box">
    <material>Gazebo/Green</material>
  </gazebo>
  <joint name="pivot_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <limits effort="1000.0" velocity="3.0"/>
    <parent link="Joe_body"/>
    <child link="pivot_box"/>
    <origin rpy="0 0 0" xyz="0 -0.16 -0.12"/>
  </joint>
  <link name="pivot_wheel">
    <visual>
      <geometry>
        <cylinder length="0.03" radius="0.04"/>
      </geometry>
      <origin rpy="0 1.57 0" xyz="0 0 0"/>
    </visual>
    <inertial>
      <mass value="3.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <collision>
      <geometry>
        <cylinder length="0.03" radius="0.04"/>
      </geometry>
      <origin rpy="0 1.57 0" xyz="0 0 0"/>
    </collision>
  </link>
  <gazebo reference="pivot_wheel">
    <material>Gazebo/Black</material>
    <mu1 value="100.0"/>
    <mu2 value="100.0"/>
    <kp value="1000000.0"/>
    <kd value="1.0"/>
  </gazebo>
  <joint name="pivot_wheel_axis" type="continuous">
    <limits effort="1000.0" velocity="50.0"/>
    <axis xyz="1 0 0"/>
    <child link="pivot_wheel"/>
    <parent link="pivot_box"/>
    <origin rpy="0 0 0" xyz="0 -0.03 -0.05"/>
  </joint>
  <link name="tower_base">
    <inertial>
      <mass value="1.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
      <origin rpy="0 0 0" xyz="0 0 0.135"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.02 0.01 0.27"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.135"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.02 0.01 0.27"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.135"/>
    </collision>
  </link>
  <gazebo reference="tower_base">
    <material>Gazebo/Green</material>
  </gazebo>
  <joint name="tower_base_joint" type="fixed">
    <parent link="Joe_body"/>
    <child link="tower_base"/>
    <origin rpy="0 0 0" xyz="0 -0.03 0.12"/>
  </joint>
  <link name="tower_boom_1">
    <inertial>
      <mass value="1.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
      <origin rpy="0 0 0" xyz="0 0.04 0"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.02 0.08 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0.04 0"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.02 0.08 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0.04 0"/>
    </collision>
  </link>
  <gazebo reference="tower_boom_1">
    <material>Gazebo/Green</material>
  </gazebo>
  <joint name="tower_boom_joint" type="fixed">
    <parent link="tower_base"/>
    <child link="tower_boom_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.27"/>
  </joint>
  <link name="tower_boom_2">
    <inertial>
      <mass value="1.0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
      <origin rpy="0 0 0" xyz="0 0 -0.015"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.02 0.01 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.015"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.02 0.01 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.015"/>
    </collision>
  </link>
  <gazebo reference="tower_boom_2">
    <material>Gazebo/Green</material>
  </gazebo>
  <joint name="boom_boom_joint" type="fixed">
    <parent link="tower_boom_1"/>
    <child link="tower_boom_2"/>
    <origin rpy="0 0 0" xyz="0 0.08 0"/>
  </joint>
</robot>

